src/rtc.c

```diff
@@ -1026,11 +1039,14 @@ void RTC_GetAlarm(alarm_t name, uint8_t *day, uint8_t *hours, uint8_t *minutes,
        * The subsecond is the bit SS[14:0] of the ALARM SSR register (not ALARMxINR)
        * to be converted in milliseconds
        */
-      if ((initMode == MODE_BINARY_MIX) || (initMode == MODE_BINARY_ONLY)) {
+      if (initMode == MODE_BINARY_ONLY) {
+        /* read the ALARM SSR register on SS[14:0] bits --> 0x7FFF */
+        *subSeconds = (((0x7fff - RTC_AlarmStructure.AlarmTime.SubSeconds + 1) & 0x7fff) * 1000) / fqce_apre;
+      } else if (initMode == MODE_BINARY_MIX) {
```

Comment:
Why add this as it was the same before in one block?

Reply:
0x7fff is the max value but depends on prediv sync ?

Reply:
grouped